spec/Candid.md

```diff
@@ -885,63 +885,61 @@ service { <name> : <functype>; <methtype>;* } <: service { <name> : <functype'>;
 
 This subtyping is implemented during the deserialisation of Candid at an expected type. As described in [Section Deserialisation](#deserialisation), the binary value is conceptually first _decoded_ into the actual type and a value of that type, and then that value is _coerced_ into the expected type.
 
-To model this, we define, for every `t1, t2` with `t1 <: t2`, a function `C[t1<:t2] : t1 -> t2`. This function maps values of type `t1` to values of type `t2`, and is indeed total.
+To model this, we define a partial coercion function `C[t1 ~> t2] : t1 -> t2`. This function maps values of type `t1` to values of type `t2`. For every `t1, t2` with `t1 <: t2`, the function is always defined, but not every `t1, t2` has a coercion.
```

Comment:
I don’t think I would model it that way, but more like we had before (before 6e62bbf277ffbe32dfadb20e7e80814c0772dba6), with the universal set of values `V` as the input, and the only change being that in `V`, references values have type annotations.

But I can’t quite put my finger on _why_ that’s better, so maybe carry on if this doesn’t match your gut feeling.

Reply:
Also `t1 ~> t2` isn’t really a meaningful relation. There is just a partial function `C` with inputs `v : t1` and `t2`. So maybe
```
C(v : t1, t2) : t2
```
would work.

Reply:
I think they are equivalent, but you are right that inference rules are more natural here.